### PR TITLE
SASS-6598: CYA Other Expenses

### PR DESCRIPTION
**Pull request by harry-ritchie**

### CODE REVIEW

File: app/viewmodels/checkAnswers/expenses/otherExpenses/OtherExpensesDisallowableAmountSummary.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      .get(OtherExpensesPage, Some(businessId))
+      .filter(areOtherExpensesDisallowable)
+      .flatMap(_ => createSummaryListRow(answers, taxYear, businessId, userType))
+
+  private def createSummaryListRow(answers: UserAnswers, taxYear: TaxYear, businessId: BusinessId, userType: UserType)(implicit
+      messages: Messages): Option[SummaryListRow] =
+    for {
+      amount             <- answers.get(OtherExpensesAmountPage, Some(businessId))
+      disallowableAmount <- answers.get(OtherExpensesDisallowableAmountPage, Some(businessId))
+    } yield SummaryListRowViewModel(
```
Comment (Jake-Raffe) [inline app/viewmodels/checkAnswers/expenses/otherExpenses/OtherExpensesDisallowableAmountSummary.scala:47]: Can use `buildRowBigDecimal` if you wish

File: conf/messages.en [hunk 2]
```diff
@@ -927,4 +928,7 @@ otherExpensesDisallowableAmount.error.overAmount.individual = The disallowable a
 otherExpensesDisallowableAmount.error.overAmount.agent = The disallowable amount of the £{0} your client spent on other expenses must be less than £100,000,000,000
 otherExpensesDisallowableAmount.title.individual = How much of the £{0} you spent on other expenses is disallowable?
 otherExpensesDisallowableAmount.title.agent = How much of the £{0} your client spent on other expenses is disallowable?
+otherExpensesDisallowableAmount.change.hidden = Change how much spent on other expenses is disallowable
 
+otherExpensesCYA.title.individual = Check your other expenses
+otherExpensesCYA.title.agent = Check your client's other expenses
```
Comment (Jake-Raffe) [inline conf/messages.en:934]: Is this the correct apostrophe?

File: test/navigation/ExpensesNavigatorSpec.scala [hunk 3]
```diff
@@ -89,6 +91,45 @@ class ExpensesNavigatorSpec extends SpecBase {
...
+                val userAnswers = UserAnswers(userAnswersId, data)
+
+                val expectedResult = OtherExpensesDisallowableAmountController.onPageLoad(taxYear, businessId, mode)
+
+                navigator.nextPage(OtherExpensesAmountPage, mode, userAnswers, taxYear, businessId) shouldBe expectedResult
+              }
+            }
+            "all expenses were claimed as allowable" - {
+              "navigate to the OtherExpensesCYAController" in {
+                val data        = Json.obj(businessId.value -> Json.obj("otherExpenses" -> "yesAllowable"))
```
Comment (Jake-Raffe) [inline test/navigation/ExpensesNavigatorSpec.scala:108]: Can test that it goes to CYA if disallowable but is CASH journey

File: app/navigation/ExpensesNavigator.scala [hunk 4]
```diff
@@ -71,26 +72,24 @@ class ExpensesNavigator @Inject() () {
...
           businessId =>
             _ =>
               userAnswers.get(OtherExpensesPage, Some(businessId)) match {
-                case Some(OtherExpenses.YesAllowable) =>
-                  standard.routes.JourneyRecoveryController.onPageLoad()
-                case Some(OtherExpenses.YesDisallowable) =>
+                case Some(YesAllowable) =>
+                  otherExpenses.routes.OtherExpensesCYAController.onPageLoad(taxYear, businessId)
+                case Some(YesDisallowable) =>
                   journeys.expenses.otherExpenses.routes.OtherExpensesDisallowableAmountController.onPageLoad(taxYear, businessId, NormalMode)
```
Comment (Jake-Raffe) [inline app/navigation/ExpensesNavigator.scala:86]: Needs to be disallowable and accrual

File: conf/messages.en (hunk 2 above)
Comment (harry-ritchie) [reply conf/messages.en:934]: i have fixed this and the messages UTs to catch this, as they currently weren't

File: test/navigation/ExpensesNavigatorSpec.scala (hunk 3 above)
Comment (Jake-Raffe) [reply test/navigation/ExpensesNavigatorSpec.scala:108]: This comment is wrong, based on an out of date prototype

File: app/navigation/ExpensesNavigator.scala (hunk 4 above)
Comment (Jake-Raffe) [reply app/navigation/ExpensesNavigator.scala:86]: This comment is wrong, based on an out of date prototype

**Review by Jake-Raffe — APPROVED** [review]

**Pull request merged**